prime-router/src/main/kotlin/fhirengine/translation/hl7/schema/ConfigSchema.kt

```diff
@@ -188,19 +190,19 @@ abstract class ConfigSchemaElement(
         when {
             !schema.isNullOrBlank() && value.isNotEmpty() ->
                 addError("Schema property cannot be used with the value property")
-            !schema.isNullOrBlank() && valueSet.isNotEmpty() ->
+            !schema.isNullOrBlank() && valueSet.getMapValues().isNotEmpty() ->
                 addError("Schema property cannot be used with the valueSet property")
             schema.isNullOrBlank() && value.isEmpty() ->
                 addError("Value property is required when not using a schema")
         }
 
         // value sets need a value to be...set
-        if (valueSet.isNotEmpty() && value.isEmpty()) {
+        if (valueSet.getMapValues().isNotEmpty() && value.isEmpty()) {
             addError("Value property is required when using a value set")
         }
 
         // value set keys and values cannot be null
-        if (valueSet.keys.any { it == null } || valueSet.values.any { it == null }) {
+        if (valueSet.getMapValues().keys.any { it == null } || valueSet.getMapValues().values.any { it == null }) {
```

Comment:
This condition probably only applies to `InlineValueSet`s, if the table is empty, that wouldn't be a reason to call the schema invalid. On the other hand, if the table or columns given are invalid, that might be reason to call the schema invalid.

Reply:
This check isn't testing if the valueset is empty but testing whether any of the keys or values in the map are `null`, which isn't allowed. This is the case the `invalid_value_set` test schema was built for.

Reply:
Ah right, my bad, disregard this one.